kart/tabular/export.py

```diff
@@ -0,0 +1,306 @@
+import os
+import re
+
+import click
+from osgeo import gdal, ogr
+
+from kart.cli_util import (
+    call_and_exit_flag,
+    KartCommand,
+)
+from kart.completion_shared import repo_path_completer
+from kart.crs_util import make_crs
+from kart.exceptions import CrsError
+
+
+def is_writable_vector_format(driver):
+    m = driver.GetMetadata()
+    return m.get("DCAP_VECTOR") == "YES" and m.get("DCAP_CREATE") == "YES"
+
+
+def list_export_formats(ctx):
+    """List the supported export formats as reported by gdal."""
+
+    for i in range(ogr.GetDriverCount()):
+        d = ogr.GetDriver(i)
+        if is_writable_vector_format(d):
+            click.echo(f"    {d.GetName()}: {d.GetMetadataItem(gdal.DMD_LONGNAME)}")
+
+
+def get_driver_by_shortname(shortname):
+    shortname = shortname.upper()
+    for i in range(ogr.GetDriverCount()):
+        d = ogr.GetDriver(i)
+        if d.GetName().upper() == shortname:
+            if not is_writable_vector_format(d):
+                raise click.UsageError(
+                    f"Driver {shortname} does not support writing vectors"
+                )
+            return d
+    raise click.UsageError(f"Destination format {shortname} not recognized")
+
+
+def get_driver_by_ext(filename):
```

Comment:
with all these things it is possible that there may not be enough info to choose a format - e.g. a user might want to export to a `.tsv` file using the CSV driver. Also there are definitely multiple drivers in GDAL which can export the same extensions in different ways

anyway, it's probably fine for now since our GDAL is pretty minimal, but we might one day need a `--format` option to explicitly set the format

Reply:
I support `kart export my-dataset CSV:output.tsv` which would work fine for the usecase above